hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSNamesystemLock.java

```diff
@@ -159,10 +160,14 @@ public void readLockInterruptibly() throws InterruptedException {
   }
 
   public void readUnlock() {
-    readUnlock(OP_NAME_OTHER);
+    readUnlock(OP_NAME_OTHER, null);
   }
 
   public void readUnlock(String opName) {
+    readUnlock(opName, null);
+  }
+
+  public void readUnlock(String opName, Supplier<String> lockReportInfoSupplier) {
```

Comment:
I'm having a hard time finding who uses this Supplier. Where are we using this?

Reply:
I use this Supplier in the following places:
https://github.com/brfrn169/hadoop/blob/HDFS-15217/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSNamesystemLock.java#L189-L190
https://github.com/brfrn169/hadoop/blob/HDFS-15217/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSNamesystemLock.java#L298-L299

The reason why I use Supplier is that we don't always print the lock report, only when the lock interval is more than the threshold (**dfs.namenode.write-lock-reporting-threshold-ms** or **dfs.namenode.read-lock-reporting-threshold-ms**). We can do lazy building additional information with the **lockReportInfoSupplier**.